src/everest/detached/jobs/everserver.py

```diff
@@ -246,24 +267,50 @@ def get_opt_progress(
         return JSONResponse(jsonable_encoder(progress))
 
     @app.post("/" + START_EXPERIMENT_ENDPOINT)
-    def start_experiment(
+    async def start_experiment(
         config: EverestConfig,
         request: Request,
+        background_tasks: BackgroundTasks,
         credentials: HTTPBasicCredentials = Depends(security),
     ) -> Response:
         _log(request)
         _check_user(credentials)
-
-        nonlocal runner
-        if runner is None:
+        if not shared_data["started"]:
             runner = ExperimentRunner(config, shared_data, msg_queue)
             try:
-                runner.start()
+                background_tasks.add_task(runner.run)
+                shared_data["started"] = True
                 return Response("Everest experiment started")
             except Exception as e:
                 return Response(f"Could not start experiment: {e!s}", status_code=501)
         return Response("Everest experiment is running")
 
+    @app.websocket("/events")
+    async def websocket_endpoint(websocket: WebSocket) -> None:
+        subscriber_id = str(uuid.uuid4())
+        await websocket.accept()
+        while True:
+            event = await get_event(subscriber_id=subscriber_id)
+            await websocket.send_json(jsonable_encoder(event))
+            await asyncio.sleep(0.1)
+            if isinstance(event, EndEvent):
+                # Give some time for subscribers to get events
+                await asyncio.sleep(5)
+                break
+
+    async def get_event(subscriber_id: str) -> StatusEvents:
+        if subscriber_id not in shared_data["subscribers"]:
+            shared_data["subscribers"][subscriber_id] = Subscriber()
+        subscriber = shared_data["subscribers"][subscriber_id]
+
+        while subscriber.index >= len(shared_data["events"]):
+            await subscriber.wait_for_event()
+
+        event = shared_data["events"][subscriber.index]
+        shared_data["subscribers"][subscriber_id].index += 1
```

Comment:
This `hared_data["subscribers"][subscriber_id].index += 1`  can be `subscriber.index  += 1` like you used it here 
```
while subscriber.index >= len(shared_data["events"]):
```